diff --git a/src/com/google/common/css/compiler/commandline/ClosureCommandLineCompiler.java b/src/com/google/common/css/compiler/commandline/ClosureCommandLineCompiler.java
index 835155c..42828ee 100644
--- a/src/com/google/common/css/compiler/commandline/ClosureCommandLineCompiler.java
+++ b/src/com/google/common/css/compiler/commandline/ClosureCommandLineCompiler.java
@@ -143,6 +143,10 @@ public class ClosureCommandLineCompiler extends DefaultCommandLineCompiler {
         "Allow unrecognized properties.")
     private boolean allowUnrecognizedProperties = false;
 
+    @Option(name = "--no-eliminate-dead-styles", usage =
+        "Prevents elimination of dead styles/duplicates/etc.")
+    private boolean noEliminateDeadStyles = false;
+
     @Option(name = "--vendor", usage =
         "Creates browser-vendor-specific output by stripping all proprietary "
         + "browser-vendor properties from the output except for those "
@@ -199,7 +203,7 @@ public class ClosureCommandLineCompiler extends DefaultCommandLineCompiler {
       builder.setProcessDependencies(true);
       builder.setExcludedClassesFromRenaming(excludedClassesFromRenaming);
       builder.setSimplifyCss(true);
-      builder.setEliminateDeadStyles(true);
+      builder.setEliminateDeadStyles(!noEliminateDeadStyles);
       builder.setCssSubstitutionMapProvider(renamingType
           .getCssSubstitutionMapProvider());
       builder.setCssRenamingPrefix(cssRenamingPrefix);
